lib/cloud_controller/organization_quota_usage_calculator.rb

```diff
@@ -0,0 +1,42 @@
+module VCAP::CloudController
+  class OrganizationQuotaUsageCalculator
+    def self.get_current_routes(org)
+      VCAP::CloudController::Route.dataset.join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_service_instances(org)
+      VCAP::CloudController::ServiceInstance.dataset.join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_domains(org)
+      VCAP::CloudController::Domain.dataset.
+        where(owning_organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_service_keys(org)
+      VCAP::CloudController::ServiceKey.dataset.join(:service_instances, id: :service_instance_id).
+        join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_ports(org)
+      VCAP::CloudController::Route.dataset.join(:spaces, id: :space_id).
```

Comment:
You can reuse `OrganizationReservedRoutePorts.new(org).count` as it is done by callers of `MaxReservedRoutePortsPolicy`.